openstudiocore/src/model/FluidCoolerTwoSpeed.hpp

```diff
@@ -0,0 +1,218 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_FLUIDCOOLERTWOSPEED_HPP
+#define MODEL_FLUIDCOOLERTWOSPEED_HPP
+
+#include <model/ModelAPI.hpp>
+#include "StraightComponent.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+// TODO: Check the following class names against object getters and setters.
+class Connection;
+class Connection;
+class Node;
+
+namespace detail {
+
+  class FluidCoolerTwoSpeed_Impl;
+
+} // detail
+
+/** FluidCoolerTwoSpeed is a StraightComponent that wraps the OpenStudio IDD object 'OS:FluidCooler:TwoSpeed'. */
+class MODEL_API FluidCoolerTwoSpeed : public StraightComponent {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit FluidCoolerTwoSpeed(const Model& model);
+
+  virtual ~FluidCoolerTwoSpeed() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> performanceInputMethodValues();
+
+  /** @name Getters */
+  //@{
+
+  // TODO: Check return type. From object lists, some candidates are: Connection.
+  boost::optional<Connection> waterInletNode() const;
+
+  // TODO: Check return type. From object lists, some candidates are: Connection.
+  boost::optional<Connection> waterOutletNode() const;
+
+  std::string performanceInputMethod() const;
+
+  boost::optional<double> highFanSpeedUfactorTimesAreaValue() const;
+
+  bool isHighFanSpeedUfactorTimesAreaValueAutosized() const;
+
+  boost::optional<double> lowFanSpeedUfactorTimesAreaValue() const;
+
+  bool isLowFanSpeedUfactorTimesAreaValueAutocalculated() const;
+
+  double lowFanSpeedUFactorTimesAreaSizingFactor() const;
+
+  double highSpeedNominalCapacity() const;
+
+  boost::optional<double> lowSpeedNominalCapacity() const;
+
+  bool isLowSpeedNominalCapacityAutocalculated() const;
+
+  double lowSpeedNominalCapacitySizingFactor() const;
+
+  double designEnteringWaterTemperature() const;
+
+  double designEnteringAirTemperature() const;
+
+  double designEnteringAirWetbulbTemperature() const;
+
+  boost::optional<double> designWaterFlowRate() const;
+
+  bool isDesignWaterFlowRateAutosized() const;
+
+  boost::optional<double> highFanSpeedAirFlowRate() const;
+
+  bool isHighFanSpeedAirFlowRateAutosized() const;
+
+  boost::optional<double> highFanSpeedFanPower() const;
+
+  bool isHighFanSpeedFanPowerAutosized() const;
+
+  boost::optional<double> lowFanSpeedAirFlowRate() const;
+
+  bool isLowFanSpeedAirFlowRateAutocalculated() const;
+
+  double lowFanSpeedAirFlowRateSizingFactor() const;
+
+  boost::optional<double> lowFanSpeedFanPower() const;
+
+  bool isLowFanSpeedFanPowerAutocalculated() const;
```

Comment:
I should have said something on the IDD review but I try to just use the keyword autosize so that the interfaces are consistent.  I find that introducing a distinction between autosize and autocalculateable pedantic and it makes the api more complicated.